### ISSUE TITLE
Error in running helics_cli on windows

### ISSUE THREAD

**Issue by asinghal91** [GMLC-TDC/helics-cli#15]
I installed helics with gridlabd and python successfully on my windows machine and I was trying to run using helics-cli. I installed the helics-cli successfully as well. However, when I am trying to run example here: https://github.com/GMLC-TDC/HELICS-Tutorial/tree/master/tutorials/2-DistributionFederation-HELICSRunner, I am getting following error:
C:\Users\sing492\OneDrive - PNNL\Documents\Projects\HELICS-Tutorial\tutorials\2-DistributionFederation-HELICSRunner>helics run --path config.json
[INFO] Running federation Tutorial-2-DistributionFederation-HELICSRunner with 2 federates.
oserr.nim(94) raiseOSError
Error: unhandled exception: The system cannot find the file specified.
Additional info: "Requested command not found: \'set\'. OS error:" [OSError]

It appears to be only caused at windows machine probably by https://github.com/GMLC-TDC/helics-cli/blob/4da6e163aa9dfa39c010ae747b1d0ce389674ba0/src/run.nim#L39

**Comment by kdheepak** [comment]
I've released a new version [v0.4.0](https://github.com/GMLC-TDC/helics-cli/releases/tag/v0.4.0). Can you try it? I don't have a windows machine on hand right now to test this, so I can't check that this works. 

The HELICS-Tutorial repository's use of helics-cli may also be outdated. I'll look into updating this as well.

**Comment by asinghal91** [comment]
Thanks @kdheepak ! The error is gone, however, some other issue is crashing the simulation. It is probably due to wrong way of running _since I am able to run my simulation successfully without helics_cli by using a run.bat script_. I am getting the following error:

```
C:\Users\sing492\OneDrive - PNNL\Documents\Projects\orison_des\model>helics run --path cosim_runner.json
[INFO] Running federation `Cosimulation-HELICSRunner` with 2 federates.
[INFO] Running federate `PythonControllerFederate`.
[INFO] Running federate `GridLABDFederate`.
[ERROR] Something went wrong with federate `PythonControllerFederate`. Please check the log files.
```
My cosim_runner.json looks like the following:

```
{
    "broker": false,
    "federates":[
        {
            "directory":"python/",
            "exec":"python controller.py",
            "host":"localhost",
            "name":"PythonControllerFederate"
        },
        {
            "directory":"gridlabd/",
            "exec":"gridlabd with_houses.glm --verbose",
            "host":"localhost",
            "name":"GridLABDFederate"
        }
    ],
    "name":"Cosimulation-HELICSRunner"
}
```

the two log files, one for each federate, are filled with empty lines, so no way to know whats going wrong.
[PythonControllerFederate.log](https://github.com/GMLC-TDC/helics-cli/files/4832527/PythonControllerFederate.log)
[GridLABDFederate.log](https://github.com/GMLC-TDC/helics-cli/files/4832528/GridLABDFederate.log)

**Comment by kdheepak** [comment]
Can you use 

```
      "exec": "python -u controller.py",
```

instead?

**Comment by asinghal91** [comment]
Tried this, still the exact same issue.

**Comment by asinghal91** [comment]
FYI: I am using the following run.bat file and it is working successfully.

```
start /b cmd /c helics_broker --federates 2 --loglevel -1
sleep 0.1
cd gridlabd
start /b cmd /c gridlabd with_houses.glm ^>../gridlabd.log
cd ..

cd python
start /b cmd /c python controller.py ^> ../controller.log
cd ..
```

**Comment by kdheepak** [comment]
Try using this `runner.json` file?

```
{
    "broker": false,
    "federates":[
        {
            "directory":"python/",
            "exec":"python controller.py",
            "host":"localhost",
            "name":"PythonControllerFederate"
        },
        {
            "directory":"gridlabd/",
            "exec":"gridlabd with_houses.glm --verbose",
            "host":"localhost",
            "name":"GridLABDFederate"
        },
        {
            "directory":"",
            "exec":"helics_broker --federate 2",
            "host":"localhost",
            "name":"HelicsBroker"
        }
    ],
    "name":"Cosimulation-HELICSRunner"
}
```

**Comment by asinghal91** [comment]
Now, I dont get any error and it keeps running with following on the console:
```
[INFO] Running federation `Cosimulation-HELICSRunner` with 3 federates.
[INFO] Running federate `PythonControllerFederate`.
[INFO] Running federate `GridLABDFederate`.
[INFO] Running federate `HelicsBroker`.
```
all 3 log files keep increasing in size (reached 6 MB till I killed the process) however, they all are empty. I checked that gridlabd output recorder files are empty though they are created. Not sure what is happening.

**Comment by kdheepak** [comment]
The log files grew to 3 MB and then they were still empty? Can you close all open windows and reopen the log files and check that they are still empty? I can't think of any reason why the log files would grow in size but still have nothing in it. 

Were the gridlabd output recorders working as expected when you ran it without helics-cli?

**Comment by asinghal91** [comment]
yeah.. I just checked and the log files grew till 114 MB as I forgot to kill the process. They all are filled with empty lines.

yes, without helics-cli, everything is working as expected. Gridlabd recorder results are as expected.

I am not sure if this issue is only with my machine. It is not urgent for me now as I am able to run using a bat file. Let me know if you are able to reproduce the same issue on your side or any other updates on this issue.

**Comment by BilalAhmadBhatti** [comment]
Hi,
I am running into the same issue. I tried to run following example on windows: 
https://github.com/GMLC-TDC/helics-cli/tree/master/examples/pi-exchange 

The simulation crashes giving error messages as seen in the attached image. All log files are also empty!!

Thanks
Bilal
![Capture](https://user-images.githubusercontent.com/39747903/87468789-3aee4600-c5e8-11ea-940d-9639fa8aeda4.PNG)

**Comment by kdheepak** [comment]
If you want an urgent fix to this, you can use the python version here: https://github.com/GMLC-TDC/helics-cli/tree/python-master

**Comment by BilalAhmadBhatti** [comment]
> If you want an urgent fix to this, you can use the python version here: https://github.com/GMLC-TDC/helics-cli/tree/python-master

I tried running echo federation example after installing this package. It throws following error from cli.py file:
![Capture](https://user-images.githubusercontent.com/39747903/87581723-6ab05300-c6a7-11ea-9e8a-d6da84e07452.PNG)

**Comment by kdheepak** [comment]
Add a `broker` flag to the `examples/echo-federation/config.json` file. I've updated the file in the repo, so if you pull you'll get the latest.

**Comment by kdheepak** [comment]
You may also need to make some of these changes: https://github.com/GMLC-TDC/helics-cli/issues/11

If you want to make a PR to `python-master` with all your fixes that would be great!

**Comment by BilalAhmadBhatti** [comment]
Thanks Dheepak.
I pulled the latest from Git and also made changes suggested by Nadia.
Now I am getting an error from subprocess.py where it doesn't seem to locate a file:
![Capture](https://user-images.githubusercontent.com/39747903/87592295-9a675700-c6b7-11ea-869d-09abe2fbfedc.PNG)

**Comment by BilalAhmadBhatti** [comment]
Update: I made a PR and fixed the runner example (echo federation) for python_master. All changes are committed to new branch named python-dev-gemini
Here is link to PR:
https://github.com/BilalAhmadBhatti/helics-cli 

Thanks
